pkg/operator/ceph/nfs/security.go

```diff
@@ -153,7 +178,51 @@ ls --all --recursive /tmp/var/lib/sss/pipes`,
 	return init, sidecar, volumes, ganeshaMounts
 }
 
-func configVolAndMount(volSource v1.VolumeSource) (v1.Volume, v1.VolumeMount) {
+func generateKrbConfResources(r *ReconcileCephNFS, nfs *cephv1.CephNFS) (
+	init *v1.Container,
+	volume *v1.Volume, // add these volumes to the pod
+	ganeshaMount *v1.VolumeMount, // add these volume mounts to the nfs-ganesha container
+) {
+	generatedKrbConfVolName := "generated-krb5-conf"
+
+	volume = &v1.Volume{
+		Name: generatedKrbConfVolName,
+		VolumeSource: v1.VolumeSource{
+			EmptyDir: &v1.EmptyDirVolumeSource{},
+		},
+	}
+	ganeshaMount = &v1.VolumeMount{
+		Name:      generatedKrbConfVolName,
+		MountPath: "/etc/krb5.conf",
+		SubPath:   "krb5.conf",
+	}
+
+	// the init container is needed to copy the starting content from the /var/lib/sss/pipes
+	// directory into the shared sockets dir so that SSSD has the content it needs to start up
+	init = &v1.Container{
+		Name: "generate-krb5-conf",
+		Command: []string{
+			"bash", "-c",
+			`set -ex
+cat << EOF > /tmp/etc/krb5.conf
+[logging]
+default = STDERR
+
+includedir /etc/krb5.conf.rook/
+EOF
+cat /tmp/etc/krb5.conf`,
```

Comment:
you can prevent the 2nd `cat` if you use the `tee` command

Reply:
I have had problems with bash's `-e` flag not work right with `tee` for some reason, so I always use `cat`. `tee` still outputs the content even if the file itself doesn't get written I think, so the second `cat` actually verifies what's in the file versus what is supposed to be in the file.